code/game/mecha/mecha.dm

```diff
@@ -924,7 +924,7 @@
 	if(istype(obstacle, /mob))//First we check if it is a mob. Mechs mostly shouln't go through them, even while phasing.
 		var/mob/M = obstacle
 		M.Move(get_step(obstacle,src.dir))
-	else if(istype(obstacle, /obj))//Then we check for regular obstacles.
+	else if(istype(obstacle, /obj) || istype(obstacle, /turf))//Then we check for regular obstacles.
 		var/obj/O = obstacle

```

Comment:
Why is this cast?

Reply:
Looks like you can safely move this cast down to the portal bit.

Reply:
Never mind - this is a strange cast considering turfs can be passed in and runtime on the `O.anchored` check

Reply:
For some godforsaken reason this bloc isn't updating.

Reply:
Oh, it's because I'm a dumbass, got it.